packages/auth/graphql/resolvers/Session.js

```diff
@@ -33,5 +37,14 @@ module.exports = {
   },
   isCurrent (session, args, { req }) {
     return session.sid === req.sessionID
+  },
+  async tokenTypes (session, args, { pgdb }) {
+    const user = await pgdb.public.users.findOne({
+      email: session.sess.email
+    })
+    const types = await Promise.all(Object.keys(TokenTypes).filter((type) =>
+      isStartable({ type: TokenTypes[type], session, pgdb, user })
+    ))
+    return [...new Set(types)]
```

Comment:
I wanted to expose which 2FA are startable so that the UI can present the appropriate component on the "Authorize" page and unauthorizedSession returns a Session object. You will never get a Session object if you don't have at least one valid token challenge.